### ISSUE TITLE
CORS Issue after 1.2.7

### ISSUE THREAD

**Issue by dasxx179** [RasaHQ/rasa#4430]
Rasa version:
rasa                               1.2.7
rasa-core                          0.14.5
rasa-core-sdk                      0.14.0
rasa-nlu                           0.15.1
rasa-sdk                           1.2.0
Rasa X version (if used & relevant):
rasa-x 0.20.2
Python version:
Python 3.7.3
python-engineio                    3.9.3
python-socketio                    4.3.1
Operating system (windows, osx, ...):
MacOs
Issue:
Hi All,
I recently opened an issue here: 
https://github.com/RasaHQ/rasa/issues/4378

It was marked to be closed (interestingly without any verification from my end), and based on the steps that I have done, the issue still remains.  
Error (including full traceback):

![image](https://user-images.githubusercontent.com/54422044/64615957-237b7680-d3a1-11e9-91b4-6c627da4ff2e.png)

![image](https://user-images.githubusercontent.com/54422044/64616037-4ad24380-d3a1-11e9-84c6-89b7454c2419.png)

Command or request that led to error:

rasa run actions --actions demo.actions //this command gives me a separate issue now:
![image](https://user-images.githubusercontent.com/54422044/64616087-6c332f80-d3a1-11e9-890b-22b71dd0fff0.png)

rasa run -m models --enable-api --cors “*” --debug 
Used docker-compose to start up the servers (I specified the specific versions of the socketio and engineio in my Dockerfile and then built an image.  Then I pulled that image in my docker-compose).  

Additionally, I would appreciate if I could verify the issue is resolved prior to closing the issue, so I do not have to create a new issue.  Thank you all!

**Comment by sara-tagger** [comment]
Thanks for the issue, @tmbo will get back to you about it soon!

###### You may find help in the [docs](https://rasa.com/docs/) and the [forum,](https://forum.rasa.com/) too 🤗

**Comment by akelad** [comment]
@dasxx179 this should work fine with `--cors "*"`. Can you show me the config of the webchat widget you're using?

**Comment by dasxx179** [comment]
> @dasxx179 this should work fine with `--cors "*"`. Can you show me the config of the webchat widget you're using?

Hi,

Gladly.  My react_component.jsx is attached below:

![image](https://user-images.githubusercontent.com/54422044/64867594-4c02ab00-d603-11e9-8620-d346c9bb68ad.png)

Then, the code for my index.html is as follows: 
![image](https://user-images.githubusercontent.com/54422044/64867842-ca5f4d00-d603-11e9-86d3-fbb6b87dc34e.png)

Please let me know if you need any additional files/info.  Thank you!

**Comment by akelad** [comment]
@wochinge can you look into testing this? I tested it on my machine before making this upgrade and it worked, so not sure what’s going on here

**Comment by wochinge** [comment]
@akelad Will do. Put it on my last for today

**Comment by wochinge** [comment]
Okay, there are actually multiple problems here.

- `no module SourcefileLoader`: can you please try the approach from https://stackoverflow.com/a/47278208/3429596 ?
- `react_component.jsx`: I think there is a error with the brackets, isn't it?
- cors error: I tried it on my machine, and it works (except that you need to pass in `--enable-api` which shouldn't be the case). Not sure whether it's a github formatting thing with the quotes around `"*"`, but could you please re-run with `rasa run -m models --enable-api --cors "*" --debug` ?

**Comment by dasxx179** [comment]
> Okay, there are actually multiple problems here.
> 
> * `no module SourcefileLoader`: can you please try the approach from https://stackoverflow.com/a/47278208/3429596 ?
> * `react_component.jsx`: I think there is a error with the brackets, isn't it?
> * cors error: I tried it on my machine, and it works (except that you need to pass in `--enable-api` which shouldn't be the case). Not sure whether it's a github formatting thing with the quotes around `"*"`, but could you please re-run with `rasa run -m models --enable-api --cors "*" --debug` ?

Hi,

I have already fixed the Sourcefileloader error with a different solution similar to what you linked.  I do not believe there is an error with the brackets because then it should not work on 1.2.2, and the command that you asked me to re-run is what I have already tried in my original post.  It is a github formatting thing, I think.  Thank you!

**Comment by firewater334** [comment]
Hi,

So it is still not working even when pulling from the latest rasa and rasa-sdk images.  However, I can get it to work by running two separate commands in two separate terminals e.g:
rasa run actions --actions demo.actions -p 5055
and
rasa run -m models --enable-api --cors "*" --debug

It is interesting how this does not work when I use these equivalent commands in docker-compose format.  

In summary, widget works fine (no CORS issue) when using the two commands separately.  Does not work when executing those same commands via docker-compose.

**Comment by firewater334** [comment]
@wochinge @akelad

**Comment by akelad** [comment]
can you send us your docker compose config?

**Comment by firewater334** [comment]
> can you send us your docker compose config?


![image](https://user-images.githubusercontent.com/47159809/65650605-3ca33a80-dfd1-11e9-8b2e-e6695d0b6666.png)
@akelad

**Comment by akelad** [comment]
@wochinge any ideas?

**Comment by firewater334** [comment]
> @wochinge any ideas?

I am not sure if this will help you, but on rasa version 1.2.2 with rasa sdk 1.2.0, it works as expected.  I do not know what modifications you guys made after those versions.

**Comment by wochinge** [comment]
the quotes for `cors` part in the docker-compose seem wrong to me.  Can you try using `"`? I can't see any way how the sdk can be connected to the issue.

**Comment by firewater334** [comment]
@wochinge I owe you an apology!  You were correct about the quotes!  Thank you very much!

**Comment by wochinge** [comment]
Awesome! Glad it works now 🚀
